tests/test_circuit_breaker.py

```diff
@@ -0,0 +1,25 @@
+import unittest
+from unittest.mock import patch
+
+from canonicalwebteam.http import BaseSession
+from canonicalwebteam.http.exceptions import CircuitOpenError
+
+
+def call_and_ignore_exceptions(fn, *args):
+    try:
+        fn(*args)
+    except Exception:
```

Comment:
By just testing for `Exception` here, you're not ensuring that it's *not* raising a `CircuitOpenError` earlier than expected (since everything is an `Exception`). It might be worth raising and checking for some other error type, like `requests.exceptions.HTTPError`.

Reply:
Good catch, I'll update!